source/Features/OverrideTonnage/Patches/MechStatisticsRules_CalculateTonnage_Patch.cs

```diff
@@ -12,6 +13,7 @@ public static void Postfix(MechDef mechDef, ref float currentValue, ref float ma
             try
             {
                 currentValue += WeightsHandler.Shared.TonnageChanges(mechDef);
+                StoredTonnage.UnRoundedTonnage = currentValue;
```

Comment:
this includes changes for
1. engine
2. gyro
3. structure
4. armor
5. chassis

i you want armor right?

also this kind of shared state is bad, as you cant associate which mechdef even has the tonnage

I think just writing a patch against `CalculateWeightChanges(BaseWeightState state, Weights weights)` would allow you to track any value yourself